File: libraryUse/stackqueue2.cpp
#include <iostream>
#include <stack>
#include <queue>
#include <complex>
using namespace std;
using point = complex<int>;

point sample_point(int i) {
  int val = 3 * i % 10;
  return point(val, 10 - val);
}

int main(void) {
  stack<point> stack;
  queue<point> queue;
  // real() の小さい順（priority queue としては real() の大きい順）
  auto cmp = [](point & p1, point & p2) { return p1.real() < p2.real(); };
  priority_queue<point, vector<point>, decltype(cmp)> priority_q(cmp);

  for (int i = 1; i < 10; i++) {
    point val = sample_point(i);
    stack.push(val);
    queue.push(val);
    priority_q.push(val);
  }

  cout << "stack:";
  while (!stack.empty()) {
    cout << " " <<  stack.top();
    stack.pop();
  }
  cout << endl;
  cout << "queue:";
  while (!queue.empty()) {
    cout << " " <<  queue.front();
    queue.pop();
  }
  cout << endl;
  cout << "priority queue:";
  while (!priority_q.empty()) {
    cout << " " <<  priority_q.top();
    priority_q.pop();
  }
  cout << endl;
  return 0;
}

File: libraryUse/StackQueue2.java
package libraryUse;

import java.awt.*;
import java.util.ArrayDeque;
import java.util.PriorityQueue;
import java.util.Queue;
import java.util.Stack;

public class StackQueue2 {
    static Point samplePoint(int i) {
        int val = 3 * i % 10;
        return new Point(val, 10 - val);
    }
    public static void main(String[] args) {
        Stack<Point> stack = new Stack<>();
        Queue<Point> queue = new ArrayDeque<>();
        // priority queue としては y の大きい順
        PriorityQueue<Point> priQ = new PriorityQueue<>((a,b)->Integer.compare(b.y, a.y));

        for (int i = 1; i < 10; i++) {
            Point val = samplePoint(i);
            stack.add(val);
            queue.add(val);
            priQ.add(val);
        }
        System.out.println("stack:");
        while(!stack.isEmpty()) {
            System.out.println(" " + stack.pop());
        }
        System.out.println();

        System.out.println("queue:");
        while(!queue.isEmpty()) {
            System.out.println(" " + queue.remove());
        }
        System.out.println();

        System.out.println("priQ:");
        while(!priQ.isEmpty()) {
            System.out.println(" " + priQ.remove());
        }
        System.out.println();
    }
}
